examples/standalone/runfile/dynamics.py

```diff
@@ -217,12 +221,13 @@ def write_global_timings(experiment, filename, comm, root=0):
 
     # collect times and output simple statistics
     comm.Barrier()
-    if not args.disable_halo_exchange:
-        print("Gathering Times")
-        experiment = set_experiment_info(
-            experiment_name, args.time_step, args.backend, args.hash
-        )
-        gather_timing_statistics(timer, experiment, comm)
+    print("Gathering Times")
+    experiment = set_experiment_info(
+        experiment_name, args.time_step, args.backend, args.hash
+    )
+    gather_timing_statistics(timer, experiment, comm)
```

Comment:
since this serves no purpose if we `disable_json_dump` I would say we should move it inside the if?

Reply:
Code refactor & gather moved down